libvast/src/system/signal_monitor.cpp

```diff
@@ -42,31 +46,33 @@ extern "C" void signal_handler(int sig) {
 
 namespace vast::system {
 
-signal_monitor_type::behavior_type
-signal_monitor(signal_monitor_type::stateful_pointer<signal_monitor_state> self,
-               std::chrono::milliseconds monitoring_interval,
-               actor receiver) {
-  VAST_DEBUG(self, "sends signals to", receiver);
+std::atomic<bool> signal_monitor::stop;
+
+void signal_monitor::run(std::chrono::milliseconds monitoring_interval,
+                         actor receiver) {
+  static constexpr auto class_name = "signal_monitor";
+  VAST_DEBUG(class_name, "sends signals to", receiver);
   for (auto s : {SIGHUP, SIGINT, SIGQUIT, SIGTERM, SIGUSR1, SIGUSR2}) {
-    VAST_DEBUG(self, "registers signal handler for", ::strsignal(s));
+    VAST_DEBUG(class_name, "registers signal handler for", ::strsignal(s));
     std::signal(s, &signal_handler);
   }
-  self->send(self, run_atom::value);
-  return {
-    [=](run_atom) {
-      if (signals[0]) {
-        signals[0] = false;
-        for (int i = 1; i < 32; ++i) {
-          if (signals[i]) {
-            VAST_DEBUG(self, "caught signal", ::strsignal(i));
-            signals[i] = false;
-            self->anon_send(receiver, signal_atom::value, i);
-          }
+  while (!stop) {
+    std::this_thread::sleep_for(monitoring_interval);
+    if (signals[0]) {
+      // TODO: this handling of singals is fundamentally unsafe, because we
+      //       always have a race between the singal handler and this loop on
+      //       singals[0]. This needs to be re-implemented in a truly atomic
+      //       fashion, probably via CAS operaions and a single 32-bit integer.
+      signals[0] = false;
+      for (int i = 1; i < 32; ++i) {
+        if (signals[i]) {
+          VAST_DEBUG(class_name, "caught signal", ::strsignal(i));
```

Comment:
Using `::` in front of `strsignal` is not necessary. I know you didn't change this, but maybe now is the time to get rid of it.